pkg/console/subresource/consoleserver/config_builder.go

```diff
@@ -317,32 +317,46 @@ func (b *ConsoleServerCLIConfigBuilder) customization() Customization {
 		conf.CustomLogoFile = b.customLogoFile
 	}
 
-	if b.devCatalogCustomization.Categories != nil {
-		mapMeta := func(meta operatorv1.DeveloperConsoleCatalogCategoryMeta) DeveloperConsoleCatalogCategoryMeta {
-			return DeveloperConsoleCatalogCategoryMeta{
-				ID:    meta.ID,
-				Label: meta.Label,
-				Tags:  meta.Tags,
-			}
-		}
+	if (!reflect.DeepEqual(b.devCatalogCustomization, operatorv1.DeveloperConsoleCatalogCustomization{})) {
 
+		var catalogTypes DeveloperConsoleCatalogTypesState
 		categories := make([]DeveloperConsoleCatalogCategory, len(b.devCatalogCustomization.Categories))
```

Comment:
Can you define `categories []DeveloperConsoleCatalogCategory` here and create the list via make in the `if b.devCatalogCustomization.Categories != nil` below?